pkg/model/keycloak_deployment_test.go

```diff
@@ -492,3 +501,77 @@ func testServiceAccountDefault(t *testing.T, deploymentFunction createDeployment
 
 	assert.Equal(t, "default", serviceAccountName)
 }
+
+func testDisableDeploymentReplicasSyncingFalse(t *testing.T, deploymentFunction createDeploymentStatefulSet, deploymentFunction2 reconcileRHSSODeployment) {
```

Comment:
After the latest changes, I noticed we have two pairs of basically identical test functions here:
* `testDisableDeploymentReplicasSyncingFalse` and `testDisableRHSSOReplicasSyncingFalse`
* `testDisableDeploymentReplicasSyncingTrue` and `testDisableRHSSOReplicasSyncingTrue`

Is there a reason for this? Maybe I'm just missing some context here. :) Could we possibly keep just `testDisableDeploymentReplicasSyncingFalse` and `testDisableDeploymentReplicasSyncingTrue` and use them both in `keycloak_deployment_test` and `rhsso_deployment_test`?

Other than that LGTM. 👍

Reply:
good catch! Thanks @vmuzikar 
Updated + squashed the commits. 
Thank you